blog: architecture driven auth part 2 (TMB)

sixhobbits opened the pull request.

RichardJECooke committed:
Draft outline

RichardJECooke committed:
english

RichardJECooke committed:
save

RichardJECooke committed:
explaining tmb security

RichardJECooke committed:
security and dpop done

RichardJECooke committed:
added when to use tmb

RichardJECooke committed:
summary done

RichardJECooke committed:
start the backend

RichardJECooke committed:
work on setting up demo

RichardJECooke committed:
run the example app and screenshots

RichardJECooke committed:
code discussed. last step is explaining the tmb flow

RichardJECooke committed:
save

RichardJECooke committed:
start mermaid diagram

RichardJECooke committed:
add mermaid diagram

RichardJECooke committed:
rename article. capital headings

RichardJECooke committed:
remove todos

RichardJECooke committed:
editing

RichardJECooke committed:
edit

RichardJECooke committed:
llm proofread

RichardJECooke committed:
edit

rideam committed:
updates

rideam committed:
fix

ritza-publishing-assistant[bot] committed:
Optimize images

rideam committed:
update BBOC link

bethh0rn committed:
Edit: Token-Mediating Backend: An alternative to the BFF architecture

bethh0rn committed:
Proofread: Token-Mediating Backend: An alternative to the BFF architecture

kmaida commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,232 @@
+
+Of course, if an attacker has managed to insert malicious JavaScript into your frontend, they can steal the access tokens again and again, so access token timeouts won't protect you. However, once the user closes their browser, all fraudulent requests must pause. Mobile and desktop apps are generally safer than web apps because they are less likely to use JavaScript.
+
+For a list of best practices mitigating the danger of client-side token storage, read the IETF's [Best Current Practice for OAuth 2.0 Security](https://datatracker.ietf.org/doc/html/rfc9700/), or this [summary](https://maida.kim/oauth2-best-practices-for-developers) by a FusionAuth developer.
```
```suggestion
For a list of best practices mitigating the danger of client-side token storage, read the [IETF's Best Current Practice for OAuth 2.0 Security](https://datatracker.ietf.org/doc/html/rfc9700/), or this [article summarizing OAuth 2.0 Best Practices for Developers](https://maida.kim/oauth2-best-practices-for-developers).
```

More descriptive link text.

sixhobbits committed:
Apply suggestions from code review

Co-authored-by: Kim Maida <yi.mihi@gmail.com>

RichardJECooke committed:
Remove image added in error in commit 88a84a83cf92c2a40b7c7613a6393d86f29faf7e that should have been loggedIn.webp

RichardJECooke committed:
show how the frontend sends the access token to resource servers

bethh0rn committed:
Consistency edit

nathan-contino commented:
Do we feel like this is ready for a full review? I'd like to queue this up a week or so after we ship Part 1, which will hopefully be in the next day or two. Let me know if it's ready for a deep review or if I should just take a look at a high level.

@sixhobbits pinging for visibility

jamesdanielwhitford committed:
Comment out Part 3 links until part 3 is published

jamesdanielwhitford committed:
Add TMB header image and update frontmatter

jamesdanielwhitford committed:
Update publish date to 2026-04-30

jamesdanielwhitford commented:
> Do we feel like this is ready for a full review? I'd like to queue this up a week or so after we ship Part 1, which will hopefully be in the next day or two. Let me know if it's ready for a deep review or if I should just take a look at a high level.
> 
> @sixhobbits pinging for visibility

Hey @nathan-contino. This is ready for a full review now that Part 1 has shipped.

Here's what I addressed today:

- Added a header image in the same style as Part 1 ([f2450b1](https://github.com/ritza-co/fusionauth-site/commit/f2450b15a))
- Set the publish date to 2026-04-30, about a week after Part 1 ([60ba793](https://github.com/ritza-co/fusionauth-site/commit/60ba79366)). The exact date will need updating when you're ready to merge
- Commented out the Part 3 links to avoid 404s, same treatment as Part 1 gave Part 2
- Opened https://github.com/FusionAuth/fusionauth-site/pull/4246 to uncomment the Part 2 link in the Part 1 article

Let me know if anything needs changing.

jamesdanielwhitford committed:
Fix CI failures: replace diagram components with mermaid blocks, fix absolute URLs

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+---
+publish_date: 2026-04-30
```
```suggestion
publish_date: 2026-05-07
```
Not your fault, but my review took longer than expected.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+publish_date: 2026-04-30
+title: "Token-Mediating Backend: An alternative to the BFF architecture"
+htmlTitle: "Part 2 of 3 on authentication architecture | FusionAuth"
+description: "Learn how a token-mediating backend (TMB) is less secure than a BFF and when to use it. Part 2 of 3 in the architecture-driven auth series."
```
Does it make sense to expand the BFF acronym in this description, since some users might stumble upon this without reading the prerequisite part 1?

nathan-contino replied:
```suggestion
description: "Learn how a token-mediating backend (TMB) is less secure than backend-for-frontend (BFF) and when to use a TMB. Part 2 of 3 in the architecture-driven auth series."
```
expansion as noted above, plus swapping out an ambiguous pronoun

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+## Understanding TMB architecture
+
+The TMB architecture is identical to the BFF pattern, except for one major difference: Access tokens are stored in the frontend (desktop app, mobile app, or web app) rather than the backend. (Refresh tokens are still stored on the backend.) Since the access token is available in the browser, the browser can call a resource server directly, instead of having to call the backend and get the backend ...
```
```suggestion
TMB works just like Backend-For-Frontend (BFF), with one major difference: TMB stores access tokens in the frontend (desktop, mobile, or web app). Since the access token is available in the browser, the browser can call a resource server directly, instead of proxying resource server calls through the backend.
```
Suggestion: rephrase some passive->active voice, trim words where possible, remove mention of refresh tokens (redundant, after mentioning that TMB = BFF except access tokens). Use 'proxying' term to trim down the final sentence.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+The TMB architecture is identical to the BFF pattern, except for one major difference: Access tokens are stored in the frontend (desktop app, mobile app, or web app) rather than the backend. (Refresh tokens are still stored on the backend.) Since the access token is available in the browser, the browser can call a resource server directly, instead of having to call the backend and get the backend ...
+
+TMB is described in the Internet Engineering Task Force (IETF) [OAuth 2.0 for Browser-Based Applications](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-token-mediating-backend) draft.
```
```suggestion
For more information about TMB, see the Internet Engineering Task Force (IETF) [OAuth 2.0 for Browser-Based Applications](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-token-mediating-backend) draft.
```
optional nitpick: make this a more obvious CTA

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+Both BFF and TMB are vulnerable to fraudulent requests if an attacker manages to inject malicious JavaScript into the browser (client hijacking). But BFF prevents the attacker from using the token from the attacker's computer (token exfiltration) while TMB does not.
+
+You can think of the security architecture spectrum like so: BFF (most secure) > TMB (less secure) > BBOC (least secure).
```
```suggestion
You can think of the security architecture spectrum like so: BFF (most secure) > TMB (less secure) > BBOC (least secure).

Both BFF and TMB are vulnerable to fraudulent requests if an attacker manages to inject malicious JavaScript into the browser (client hijacking). But BFF prevents the attacker from using the token from the attacker's computer (token exfiltration) while TMB does not.
```
Consider swapping the order of these paragraphs. I think the spectrum visualization helps the reader better understand the detailed comparison on first scan.

nathan-contino replied:
This comment applies globally: sometimes you prefix BFF/TMB/BBOC with "a ", sometimes you don't. We should make this treatment consistent (IMO we're better off without the article, but I trust your decision on this)

jamesdanielwhitford replied:
agreed on dropping the article. went through and removed "a"/"an" before BFF, TMB, and BBOC everywhere in the doc for consistency. done in commit 8fc6f5d.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+- Same-site HTTP-only session cookies (as well as PKCE checks and OIDC nonces) prevent CSRF attacks.
+- Refresh tokens stay on the server, hidden from attackers.
+
+In a TMB frontend, the access token in the browser can't be kept in an HTTP-only cookie, because JavaScript needs to access it to send it in calls to resource servers (which have different URL domains to the authorization server domain). Since the access token has to be stored in the browser's local storage or application memory, it's vulnerable to exfiltration by XSS attacks, like malicious npm  ...
```
three "it"s in a single paragraph is a bit much. Could we rewrite this to be less confusing?


```suggestion
In a TMB frontend, the access token can't be kept in an HTTP-only cookie, because the browser needs the token for calls to resource servers. Since the access token has to be stored in the browser's local storage or application memory, it's vulnerable to exfiltration by XSS attacks, like malicious npm packages or unsanitized HTML inputs.
```

I think this conveys the same point in a lot less words and pronouns, but feel free to tweak it if I've misconstrued.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+You can set a very short expiry period for your access tokens (ten minutes or less) to minimize the attack window. The frontend can silently ask the server for a new access token generated from the server-side refresh token every few minutes, so the user does not need to log in again. If you detect unauthorized use of an access token, you can revoke the refresh token and force a user to log in ag ...
+
+Of course, if an attacker has managed to insert malicious JavaScript into your frontend, they can steal the access tokens again and again, so access token timeouts won't protect you. However, once the user closes their browser, all fraudulent requests must pause. Mobile and desktop apps are generally safer than web apps because they are less likely to use JavaScript.
```
```suggestion
Of course, if an attacker has managed to insert malicious JavaScript into your frontend, they can steal the access tokens again and again, so access token timeouts won't protect you. However, once the user closes their browser, all fraudulent requests must pause. For this reason, mobile and desktop apps are safer than web apps for TMB.
```
IMO it's not javascript that's the problem, it's the Wild West browser environment where ads and other scripts on a page can do whatever they want.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+### TMB with DPoP
+
+There is one relatively new technology (2023) that makes client-side tokens a lot safer — the Demonstrating Proof of Possession (DPoP) mechanism, available as the IETF [RFC 9449](https://datatracker.ietf.org/doc/rfc9449).
```
```suggestion
One relatively new technology makes client-side tokens a lot safer — Demonstrating Proof of Possession (DPoP), available as the IETF [RFC 9449](https://datatracker.ietf.org/doc/rfc9449).
```
suggestion: trim some words

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+The security ranking is: BFF = TMB with DPoP ≫ TMB > BBOC.
+
+DPoP is not yet supported by FusionAuth, but it has been [requested](https://github.com/FusionAuth/fusionauth-issues/issues/1679). You can learn more about exactly how DPoP works in the [Evolving Auth Standards: Security Best Practices for App Developers webinar](/webinar/evolving-auth-standards-security-best-practices-for-app-developers).
```
```suggestion
Before DPoP, if an attacker got their hands on your OAuth access token, they could use it to make requests just like you. With DPoP, the token alone isn't sufficient; the client must also prove their identity with a cryptographic signature.

DPoP works by binding the token to a DPoP key, then requiring both the access token and a signature in requests. DPoP-enabled APIs reject any request that isn't signed by the DPoP key.

Your browser stores the DPoP key using [a web cryptography API that prevents exfiltration](https://developer.mozilla.org/en-US/docs/Web/API/CryptoKey/extractable). The browser can sign requests with the DPoP key, but it can't access the key itself.

An attacker that manages to insert malicious JavaScript into your browser could make calls using a token signed by the key. But the token is only useful when the attacker can sign requests with the DPoP key, which is only possible in your browser. So we end up with a risk profile equivalent to BFF:

BFF == TMB with DPoP > TMB > BBOC

FusionAuth added support for DPoP in [version `1.63.0`](/blog/announcing-fusionauth-1-63). For a great visual walkthrough of the protocol, see the [interactive DPoP site](https://dpop.info).
```
Suggestion: ditch the analogy (not a fan of comparing cryptographic signatures to real-world signatures), add reference to the FA version that added DPoP, tighten up language that keeps using 'access code' when I think we mean access token, introduce the security comparison in a more organic way, update the comparison with developer-friendly comparators

jamesdanielwhitford replied:
applied your rewrite. done in commit 79c80975.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+htmlTitle: "Part 2 of 3 on authentication architecture | FusionAuth"
+description: "Learn how a token-mediating backend (TMB) is less secure than a BFF and when to use it. Part 2 of 3 in the architecture-driven auth series."
+image: /img/blogs/bff/tmb-security-architecture-header.png
+authors: Kim Maida
```
```suggestion
authors: Kim Maida, Nathan Contino
```
Since I'm doing a heavier edit of this post, I figure I might as well

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+This section explains the TMB OAuth flow in more detail. It is based on the [official text from the IETF](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-application-architecture-2).
+
+Below is the sequence if the browser loads the app and still has a session cookie from the previously active session.
```
```suggestion
The following sequence shows the browser loading the app with a session cookie from a previously active session:
```

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
````diff
@@ -0,0 +1,280 @@
+    B->>R: Make a purchase (include access token)
+```
+
+If there is no session, the user has to log in. This sequence is shown below.
````
```suggestion
When no previous session cookie exists, the user must first log in:
```

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
````diff
@@ -0,0 +1,280 @@
+
+```sh
+docker compose up -d
+```
````
````suggestion
1. Open a terminal and clone the repository with Git using the commands below:

   ```console
   git clone https://github.com/kmaida/auth-architecture.git
   ```

   ```console
   cd auth-architecture
   ```

   ```console
   cp .env.sample .env
   ```

1. Start the FusionAuth server with Docker:

   ```console
   docker compose up -d
   ```
````
suggestion: indent code samples within their parent steps. mark interactive shell snippets as 'console' so it's more intuitive to copy/paste them into a shell. break shell snippets into copy-pasteable one liners to take advantage of our copy snippet button.

jamesdanielwhitford replied:
applied. indented all code blocks within their steps, switched from `sh` to `console`, split into one-liners. done in commit 79c80975.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
````diff
@@ -0,0 +1,280 @@
+docker compose up -d
+```
+
+This should download Docker images and start FusionAuth on `http://localhost:9011/admin`. You can log in with the default admin credentials provided in the README: `admin@example.com` and `password`.
````
```suggestion
This should download Docker images and start FusionAuth on [http://localhost:9011/admin](http://localhost:9011/admin). You can log in with the default admin credentials provided in the README: `admin@example.com` and `password`.
```
Might as well link to make this easier for our users.

jamesdanielwhitford replied:
applied your suggestion and linked both localhost URLs. just flagging a conflict with the contributing guide: [CONTRIBUTING.md line 22](https://github.com/FusionAuth/fusionauth-site/blob/main/CONTRIBUTING.md#L22) says to remove hyperlinks to `localhost`. happy to revert if you'd prefer to follow the existing rule, or leave as-is if the contributing doc should be updated.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
````diff
@@ -0,0 +1,280 @@
+```js
+CLIENT_ID="e72dca1d-626c-4f4b-8f36-b7c8c2c0af33"
+CLIENT_SECRET="TC3Kmq9yNgudIHl8BKLJXJFAhd8AmzfTwjJSqAFJJ-k"
+```
````
````suggestion
1. Run the following commands to prepare the backend settings file for editing:

   ```console
   cd tmb/backend
   ```

   ```console
   cp .env.sample .env
   ```

1. Open the `tmb/backend/.env` file in a text editor and update the following values with your own Client ID and secrets, found in the FusionAuth Admin UI under <BreadCrumb>Applications -> Your App -> Action -> Edit</Breadcrumb>:

   ```dotenv
   CLIENT_ID="e72dca1d-626c-4f4b-8f36-b7c8c2c0af33"
   CLIENT_SECRET="TC3Kmq9yNgudIHl8BKLJXJFAhd8AmzfTwjJSqAFJJ-k"
   ```
````
Same as above re:snippets.

I believe we should use `dotenv` as the language for `.env` files -- javascript is probably not correct!

Let's use breadcrumb annotation here. Please add the import at the top of the file (or all hell will break loose). IMO we're better off explaining where to find these values before the snippet, instead of in an aside that lots of readers might skip.

jamesdanielwhitford replied:
applied your suggestion. one small deviation: changed `Admin UI` to `admin UI` per [CONTRIBUTING.md line 14](https://github.com/FusionAuth/fusionauth-site/blob/main/CONTRIBUTING.md#L14). done in commit 3311f7ca. let me know if you'd prefer to keep it capitalised.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+  <Aside type="note">
+    You can also find the values in the FusionAuth admin UI under **Applications -> Your App -> Action -> Edit**.
+  </Aside>
```
```suggestion
```

Remove this since we're moving these instructions into the step text.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
````diff
@@ -0,0 +1,280 @@
+```sh
+npm install
+npm run dev
+```
````
````suggestion
1. Start the server with the following commands:

   ```console
   npm install
   ```

   ``console
   npm run dev
   ```
````
s/command/commands and indent/update the language for these snippets

jamesdanielwhitford replied:
applied. split into one-liners, indented, switched to `console`. done in commit 79c80975.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+  <Aside type="note">
+    The login page is served by FusionAuth, not by the app, but both pages have matching visual styles.
+  </Aside>
```
````suggestion
1. In a new terminal in the repository directory, run the following commands to prepare the frontend settings file:

   ```console
   cd tmb/frontend
   ```

   ```console
   cp .env.sample .env
   ```

1. Start the server with the following commands:

   ```console
   npm install
   ```

   ```console
   npm run dev
   ```

   Now you have an authentication provider, backend API, and frontend website running and you can test the full system.

1. Browse to the frontend at [http://localhost:5173](http://localhost:5173).

   ![TMB website homepage when logged out](/img/blogs/auth-architecture/loggedOut.png)

1. Click the login button at the top right and log in with the test user `user@example.com` and the password `password`.

   ![TMB website login page](/img/blogs/auth-architecture/logIn.png)
````
Similar comments to above, with indenting images into steps this time, and removing an aside that I think doesn't add much to the narrative.

jamesdanielwhitford replied:
applied. split into one-liners, indented code blocks and images into steps, switched to `console`, removed the aside. done in commit 79c80975.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
```diff
@@ -0,0 +1,280 @@
+
+**The authentication flow** is hand-coded in this example to make it understandable and explicit, but in reality you should use a library like [Passport.js](https://www.passportjs.org/packages/passport-oauth2) to do most of the OAuth 2 work. Passport handles only the interaction between your backend and FusionAuth, not between your backend and your frontend. The important part of TMB that Passpor ...
+
+**The frontend project** is very simple. Its only dependency is the React.js ecosystem, and authentication is handled in `services/AuthContext.jsx`. Since the FusionAuth webpage redirect handles login, the only authentication work the frontend needs to do is check whether the user is logged in and get new access tokens.
```
```suggestion
**The frontend project** is very simple. Its only dependency is the React.js ecosystem, and authentication is handled in `services/AuthContext.jsx`. FusionAuth provides an access token, and the frontend uses the access token to call APIs directly, without a proxy through the backend, both for same-origin and cross-origin APIs.
```
I think this adds the extra detail Kim was hoping for?

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-TMB.mdx:
````diff
@@ -0,0 +1,280 @@
+```js
+fetch(`${resourceApiUrl}/api/recipe`,
+  {headers: {'Authorization': `Bearer ${accessToken}`,
+```
````
I was under the impression that TMB _doesn't_ need to forward or proxy the access token? Regardless, I think we can cut the first section of this, since it's now covered by the frontend bullet point above:


````suggestion
You can see the TMB API call logic in [`ResourceApiPage.jsx`](https://github.com/kmaida/auth-architecture/blob/main/tmb/frontend/src/pages/ResourceApiPage.jsx). The `fetchRecipe()` function passes the token to the resource server in the header of fetch calls:
```js
fetch(`${resourceApiUrl}/api/recipe`,
  {headers: {'Authorization': `Bearer ${accessToken}`,
```
````

jamesdanielwhitford replied:
cloned kmaida/auth-architecture and checked ResourceApiPage.jsx directly. TMB does forward the access token: fetchRecipe() calls getAccessToken() and sends it as Authorization: Bearer ${accessToken} directly to the resource server. so the original sentence was technically correct, but agreed it was redundant given the updated frontend bullet above. cut it and replaced with "You can see the TMB API call logic in ResourceApiPage.jsx..." per your suggestion. done in commit 2931134b.

nathan-contino commented on astro/src/content/blog/auth-architecture-part2-tmb.mdx:
```diff
@@ -0,0 +1,280 @@
+
+The recommendation of this article, and the whole series, is to use BFF for authentication and authorization. BFF is far more secure than TMB and BBOC. You can only justify using TMB instead of a secure backend if your app needs near real-time speed or if you have an insurmountable infrastructure constraint.
+
+If your authentication provider doesn't support BFF, you can implement one using FusionAuth for free, either as the Identity Provider itself or as merely the gateway that securely proxies authentication. See the [BFF tutorial](/blog/backend-for-frontend#use-a-premade-bff) to learn how.
```
```suggestion
If your authentication provider doesn't support BFF, you can implement one using FusionAuth for free, either as the Identity Provider itself or as merely the gateway that securely proxies authentication. See the [BFF tutorial](/blog/backend-for-frontend#use-a-premade-bff) to learn how.

And if you absolutely must use TMB, combine it with [DPoP (also supported by FusionAuth!)](/docs/lifecycle/authenticate-users/oauth/dpop) to reduce your risk profile.
```

nathan-contino commented on astro/src/diagrams/blog/auth-architecture-part2-TMB/browser-no-session.astro:
```diff
@@ -0,0 +1,31 @@
+`;
+---
+<Diagram code={diagram} alt={alt}/>
+
```
We can delete both of these external files now that our documentation supports inline mermaid fences.

jamesdanielwhitford replied:
done. confirmed no remaining references to either file in the MDX before deleting. done in commit 8c4f67bc.

nathan-contino reviewed (changes requested):
Looking pretty good, will LGTM if you implement all/most of my suggestions. Hoping we can get this out 5/7, it's quite close to ready!

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Update astro/src/content/blog/auth-architecture-part2-TMB.mdx

Co-authored-by: nathan contino <patronus@lambdalatitudinarians.org>

jamesdanielwhitford committed:
Restructure setup section: indent code blocks, console language, split one-liners, remove aside

jamesdanielwhitford committed:
Add Breadcrumb import and component for admin UI navigation path

jamesdanielwhitford committed:
Fix no-explicit-any ESLint errors in getStaticIndexPaths

jamesdanielwhitford committed:
Fix no-explicit-any ESLint errors in getStaticIndexPaths

jamesdanielwhitford committed:
Remove redundant access token forwarding sentence before ResourceApiPage snippet

jamesdanielwhitford committed:
Delete dead browser-no-session and browser-active-session astro diagram files

jamesdanielwhitford committed:
Remove articles before BFF/TMB/BBOC acronyms for consistent treatment

jamesdanielwhitford committed:
Style guide pass: headers to title case, admin UI casing, alt text, article treatment

jamesdanielwhitford committed:
Remove colons before code snippets per style guide

jamesdanielwhitford committed:
Rename blog post filename to lowercase for consistent URL routing

jamesdanielwhitford committed:
Restore colons before code snippets, update Admin UI casing per Nathan's feedback

- Update CONTRIBUTING.md: replace no-colon rule with guidance allowing colons or periods before snippets
- Update CONTRIBUTING.md: change "admin UI" to "Admin UI"
- Restore colons before all code snippet intros in TMB article
- Update "admin UI" to "Admin UI" in TMB article

nathan-contino reviewed (approved):
LGTM!
